jepsen/src/jepsen/control.clj

```diff
@@ -187,12 +188,25 @@
   (rc/with-conn [s *session*]
     (ssh/scp-to *session* current-path node-path)))
 
+(defn file->path
+  "Takes an object, if it's an instance of java.io.File, gets the path, otherwise
+  returns the object"
+  [x]
+  (if (instance? java.io.File x)
+    (.getCanonicalPath x)
+    x))
+
 (defn upload
-  "Copies local path(s) to remote node. Takes arguments for clj-ssh/scp-to."
-  [& args]
+  "Copies local path(s) to remote node and returns the remote path.
+  Takes arguments for clj-ssh/scp-to."
+  [& [local-paths remote-path & remaining]]
```

Comment:
I'm doing some funky comboed destructuring + & rest-ing here, if there's a better way to do this I'm open to refactoring it